### PR TITLE
feat(search): search on ProfilesList

### LINKED ISSUE #1005: Search engine on list of profiles pages
Main story : If I discovered DisMoi through a profile I want find it back after I installed DisMoi

**Remarque 1 :**  Si l'utilisateur recherche de façon libre, au moment où il valide, la page de résultat chargée décoche automatiquement tous les filtres 
CàD pas de combinaison entre filtre et recherche libre

**Remarque 2 :**  Pas de recherche libre sur mobile pour l'instant

### PR DESCRIPTION
**Pull request by gregoirelacoste**
closes #1005 
I did a first try, you can test and tell me.
the search engine use :
- filters
- the titles
- the intros

with AND logic

there is no UI for now

@newick

### CODE REVIEW

**Comment by newick** [comment]
wooow nice start!
@gregoirelacoste what's next step? Me to add some UI or do you have some tech work to do?

**Comment by gregoirelacoste** [comment]
> Finally no search button?
indeed, no need it's live searching
tech is fine, so you can work on UI, or I can if you want

**Comment by newick** [comment]
image
Added some design from Marco.

**Comment by newick** [comment]
@gregoirelacoste is there a way to have the filters not in capitalize? I removed it in the code but I don't see them in lowercase.

**Comment by gregoirelacoste** [comment]
> @gregoirelacoste is there a way to have the filters not in capitalize? I removed it in the code but I don't see them in lowercase.

Why do tout need that ?

**Comment by newick** [comment]
@gregoirelacoste 
image
no uppercase :)

**Comment by gregoirelacoste** [comment]
@newick everything is good for you here ?

**Comment by gregoirelacoste** [comment]
@JalilArfaoui @lutangar hi, I need this feature to finish this one #1009 could you review it please ?
thx!

**Comment by newick** [comment]
@gregoirelacoste lgtm!

File: src/app/profiles/App/Pages/Profiles/List/ProfileList.tsx [hunk 1]
```diff
@@ -22,6 +21,8 @@ import NotConnectedPopin, {
 } from '../NotConnectedPopin';
 import onContributorExampleClick from '../../../onContributorExampleClick';
 import { ContextPopinState } from '../../../../store/reducers/contextPopin.reducer';
+import { Input, Select } from '../../../../../../components/atoms/Forms';
```
Comment (lutangar) [inline src/app/profiles/App/Pages/Profiles/List/ProfileList.tsx:24]: Please, prefer absolute imports.

File: src/app/profiles/App/useContributorsFilters.ts [hunk 2]
```diff
@@ -1,43 +1,55 @@
...
+import React, { useState, useEffect } from 'react';
 import { StatefulContributor } from 'libs/domain/contributor';
-import { Categories } from 'libs/domain/category';
-
-export const distinct = (value: string, index: number, array: string[]) =>
-  array.indexOf(value) === index;
+import { ALL } from 'components/molecules/Filters/RadiosFilters';
+import getSearch from '../../../libs/utils/getSearch';
 
 function useContributorsFilters(
```
Comment (lutangar) [inline src/app/profiles/App/useContributorsFilters.ts:6]: This file should be with other hooks in `app/profiles/hooks`

File: src/libs/utils/getSearch.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { StatefulContributor } from '../domain/contributor';
+
+const getSearch = (
```
Comment (lutangar) [inline src/libs/utils/getSearch.ts:3]: Should probably be named something like `getSearchResults` instead

File: src/app/profiles/App/useContributorsFilters.ts [hunk 4]
```diff
@@ -1,43 +1,55 @@
-import { useState, useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
 import { StatefulContributor } from 'libs/domain/contributor';
-import { Categories } from 'libs/domain/category';
-
-export const distinct = (value: string, index: number, array: string[]) =>
-  array.indexOf(value) === index;
+import { ALL } from 'components/molecules/Filters/RadiosFilters';
+import getSearch from '../../../libs/utils/getSearch';
```
Comment (lutangar) [inline src/app/profiles/App/useContributorsFilters.ts:4]: > Please, prefer absolute imports.

**Pull request merged**